From e8e1b519d6d4930ee74f1df4c822cd19abf61d12 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 5 Jan 2023 11:52:03 +0200
Subject: [PATCH] imx8mm_solidrun: tweak calibration parameters for 1gb samsung
 ddr

The previous calibhration fails running memtester with the refresh cycle
time as specified in datasheet.
Raise per-bank refresh cycle time to 180ns for stable operation.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../imx8mm_solidrun/lpddr4_samsung_K4F8E3S4HD-MGCL.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/board/solidrun/imx8mm_solidrun/lpddr4_samsung_K4F8E3S4HD-MGCL.c b/board/solidrun/imx8mm_solidrun/lpddr4_samsung_K4F8E3S4HD-MGCL.c
index 2e45494478..94e6bf4bf0 100755
--- a/board/solidrun/imx8mm_solidrun/lpddr4_samsung_K4F8E3S4HD-MGCL.c
+++ b/board/solidrun/imx8mm_solidrun/lpddr4_samsung_K4F8E3S4HD-MGCL.c
@@ -21,7 +21,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d400000, 0xa1080020 },
 	{ 0x3d400020, 0x223 },
 	{ 0x3d400024, 0x3a980 },
-	{ 0x3d400064, 0xb0060 },
+	{ 0x3d400064, 0xb0087 },
 	{ 0x3d4000d0, 0xc00305ba },
 	{ 0x3d4000d4, 0x940000 },
 	{ 0x3d4000dc, 0xd4002d },
@@ -37,7 +37,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d40011c, 0x402 },
 	{ 0x3d400130, 0x20600 },
 	{ 0x3d400134, 0xc100002 },
-	{ 0x3d400138, 0x66 },
+	{ 0x3d400138, 0x8d },
 	{ 0x3d400144, 0x96004b },
 	{ 0x3d400180, 0x2ee0017 },
 	{ 0x3d400184, 0x2605b8e },
@@ -74,7 +74,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d402020, 0x21 },
 	{ 0x3d402024, 0x7d00 },
 	{ 0x3d402050, 0x20d040 },
-	{ 0x3d402064, 0x3000d },
+	{ 0x3d402064, 0x30012 },
 	{ 0x3d4020dc, 0x840000 },
 	{ 0x3d4020e0, 0x310000 },
 	{ 0x3d4020e8, 0x66004d },
@@ -89,7 +89,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d40211c, 0x302 },
 	{ 0x3d402130, 0x20300 },
 	{ 0x3d402134, 0xa100002 },
-	{ 0x3d402138, 0xe },
+	{ 0x3d402138, 0x13 },
 	{ 0x3d402144, 0x14000a },
 	{ 0x3d402180, 0x640004 },
 	{ 0x3d402190, 0x3818200 },
@@ -99,7 +99,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d403020, 0x21 },
 	{ 0x3d403024, 0x1f40 },
 	{ 0x3d403050, 0x20d040 },
-	{ 0x3d403064, 0x30004 },
+	{ 0x3d403064, 0x30005 },
 	{ 0x3d4030dc, 0x840000 },
 	{ 0x3d4030e0, 0x310000 },
 	{ 0x3d4030e8, 0x66004d },
@@ -114,7 +114,7 @@ static struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d40311c, 0x302 },
 	{ 0x3d403130, 0x20300 },
 	{ 0x3d403134, 0xa100002 },
-	{ 0x3d403138, 0x4 },
+	{ 0x3d403138, 0x5 },
 	{ 0x3d403144, 0x50003 },
 	{ 0x3d403180, 0x190004 },
 	{ 0x3d403190, 0x3818200 },
-- 
2.35.3

